apps/web/pages/api/book/event.ts

```diff
@@ -6,11 +6,31 @@ import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowE
 import getIP from "@calcom/lib/getIP";
 import { checkCfTurnstileToken } from "@calcom/lib/server/checkCfTurnstileToken";
 import { defaultResponder } from "@calcom/lib/server/defaultResponder";
+import type { TraceContext } from "@calcom/lib/tracing";
+import { distributedTracing } from "@calcom/lib/tracing/factory";
 import { CreationSource } from "@calcom/prisma/enums";
 
-async function handler(req: NextApiRequest & { userId?: number }) {
+async function handler(req: NextApiRequest & { userId?: number; traceContext?: TraceContext }) {
   const userIp = getIP(req);
 
+  const traceContext = req.traceContext
+    ? distributedTracing.updateTrace(req.traceContext, {
+        userIp,
+        eventTypeId: req.body?.eventTypeId,
+      })
+    : distributedTracing.createTrace("api_book_event", {
+        meta: {
+          userIp,
+          eventTypeId: req.body?.eventTypeId,
+        },
```

Comment:
Why is it conditional? Why don\t we make traceContext required and as discussed the midddleware and defaultHandler would provide it.